### ISSUE TITLE
inconsistent semicolon expansion in catch_discover_tests

### ISSUE THREAD

**Issue by georg-emg** [catchorg/Catch2#2214]
When using the cmake command `catch_discover_tests` on a target with a property `CROSSCOMPILING_EMULATOR`, catch2 will use the emulator to discover and execute tests. If one of the parameters of the cross compiling emulator contains an escaped semicolon, however, the behaviour is inconsistent between the test discovery and the test execution. When running the test app to discover the tests, escaped semicolons are correctly turned into plain semicolons. When the tests are then run using ctest, however, escaped semicolons are falsely interpreted as separators between individual command line parameters.

Example:

consider the following cmake code fragment for a Windows test app:
```cmake
add_executable(myTestApp test.cpp)
set_target_properties(myTestApp PROPERTIES CROSSCOMPILING_EMULATOR "${CMAKE_COMMAND};-E;env;PATH=C:\SomeDirectory\;C:\SomeOtherDirectory")
catch_discover_tests(myTestApp)
```
This will run myTestApp.exe as:

```
cmake.exe -E env PATH=C:\SomeDirectory;C:\SomeOtherDirectory myTestApp.exe --list-test-names-only
```

however, the generated include file `myTestApp_tests-xxxxxxx.cmake`that contains the cmake tests, will contain lines in the form:

```cmake
add_test(some_test_name cmake.exe -E env [==[PATH=C:\SomeDirectory]==] [==[C:\SomeOtherDirectory]==] myTestApp.exe some_test_name)
#                                                                     ^ Note the wrongly split argument here
```
instead of:

```cmake
add_test(some_test_name cmake.exe -E env [==[PATH=C:\SomeDirectory;C:\SomeOtherDirectory]==] myTestApp.exe some_test_name)
#                                                                 ^ There should be a semicolon like shown here
```

**Comment by Minoru** [comment]
This issue is mentioned [in the changelog for 2.13.6](https://github.com/catchorg/Catch2/releases/tag/v2.13.6), but the PR that fixed it (https://github.com/catchorg/Catch2/pull/2215) is actually merged into `develop` branch, not `2.x`, so the fix is absent from 2.13.6 release. No big deal (to me), just letting you know, @horenmar.

**Comment by horenmar** [comment]
@Minoru Thanks, but the fix is merged into `v2` via #2216. However the discussion for the fix was in #2215, so I linked that one in the release notes.

**Comment by Minoru** [comment]
Ah, sorry, my bad! I failed to realize this is related to CMake, not the hpp file.
